### PR TITLE
Add a tool to request info when a bug is moved to Core::Performance

### LINKED ISSUE #1887: Automatically add a comment when a bug is moved to Core::Performance
Hey,

When a bug is moved to the Core::Performance component, [our process involves adding a comment to request additional information from the reporter](https://wiki.mozilla.org/Performance/Triage#Moved_to_Core::Performance).

Would it be possible to automatically add this comment with a needinfo to the reporter whenever a bug is moved to Core::Performance?
Here is the text:
```
This bug was moved into the Performance component. Reporter, could you make sure the following information is on this bug?

 - For slowness or high CPU usage, capture a profile with http://profiler.firefox.com/ , upload it and share the link here.
 - For memory usage issues, capture a memory dump from about:memory and attach it to this bug.
 - Troubleshooting information: Go to about:support, click "Copy raw data to clipboard", paste it into a file, save it, and attach the file here.

Thank you.
```

We were thinking of some edge cases that should be taken care of:
* if the bug is moved by mistake => the bot could wait for ~1 day before adding a comment, to allow for a user to move the bug again.
* avoid adding the comment twice (if the bug is moved back and forth)

Thanks in advance!

cc @davehunt

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Resolves #1887

## Checklist

- [x] Type annotations added to new functions
- [x] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/relman-auto-nag/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

**Comment by suhaibmujahid** [comment]
# Dry-run

<p>The following bugs were moved into the Core::Performance component; a comment to request more information was added:
<table style="border:1px solid black;border-collapse:collapse;" border="1">
<thead>
<tr>
<th>Bug</th><th>Summary</th>
</tr>
</thead>
<tbody>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1819303">1819303</a>
</td>
<td>
ark.intel.com makes Firefox consume all available memory
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1260981">1260981</a>
</td>
<td>
[Meta] Identify the cause of performance problems in using google docs
</td>
</tr>
</tbody>
</table>
</p>

## Comment example - [bug 1819303](https://bugzilla.mozilla.org/show_bug.cgi?id=1819303)

This bug was moved into the Performance component.

:kamil.paral, could you make sure the following information is on this bug?

 - For slowness or high CPU usage, capture a profile with http://profiler.firefox.com/ , upload it and share the link here.
 - For memory usage issues, capture a memory dump from `about:memory` and attach it to this bug.
 - Troubleshooting information: Go to `about:support`, click "Copy raw data to clipboard", paste it into a file, save it, and attach the file here.

Thank you.

**Comment by marco-c** [comment]
What if the information is already present in the bug?

**Comment by suhaibmujahid** [comment]
> What if the information is already present in the bug?

In that scenario the needinfo will be cleared with/without a confirmation that the information is already in the bug.

We could add a sentence at the end to request a response: "If the requested information is already in the bug, please confirm that." 


@julienw wdyt?

**Comment by julienw** [comment]
Yeah, this sounds good to me. We can adjust later if we see this case happening too much.

**Comment by marco-c** [comment]
Could we add some basic checks to prevent some obvious cases? E.g. if an attachment with "memory" in its name is in the bug, we don't ask for a memory report; if there is a "share.firefox.dev" in any of the comments, we don't ask for a performance profile.

**Comment by julienw** [comment]
This would work for me too :-)

**Comment by suhaibmujahid** [comment]
> Could we add some basic checks to prevent some obvious cases? E.g. if an attachment with "memory" in its name is in the bug, we don't ask for a memory report; if there is a "share.firefox.dev" in any of the comments, we don't ask for a performance profile.

Nice idea! However, doing simple checks could have high false negatives. I suggest monitoring the cases after deploying this and adjusting later if this introduces much noise.

I filed an issue to follow up on this idea (i.e., #1926).

@marco-c wdyt?

**Comment by marco-c** [comment]
I would add some simple checks now, the ones that are very unlikely to have false negatives and are easy to implement.
- "share.firefox.dev" in the comments is definitely a link to a profile, so it is very unlikely to cause false negatives.
- if you go to about:memory and click on "Measure and save...", Firefox will save a memory-report.json.gz file on your disk, so  "memory-report.json.gz" in the attachments is also very unlikely to cause false negatives.

**Comment by suhaibmujahid** [comment]
# Dry-run

```
2023-03-10 23:13:31,200 - INFO - Run tool moved_to_performance.py
2023-03-10 23:13:32,903 - INFO - Tool moved_to_performance.py has finished.
```

<p>The following bugs were moved into the Core::Performance component; a comment to request more information was added:
<table style="border:1px solid black;border-collapse:collapse;" border="1">
<thead>
<tr>
<th>Bug</th><th>Summary</th>
</tr>
</thead>
<tbody>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1821555">1821555</a>
</td>
<td>
Both FB Marketplace &amp; Twitter lock up for 30-45 seconds every time I try to scroll.
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1821424">1821424</a>
</td>
<td>
Memory leakage 110.0.01 Win11 22H2
</td>
</tr>
<tr bgcolor="#E0E0E0">
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1820078">1820078</a>
</td>
<td>
I try to access https://mysafaricost.com on firefox browser and it redirects always on some pages and stops and displays a broken site at the end while it opens on another browser.
</td>
</tr>
<tr >
<td>
<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=1795009">1795009</a>
</td>
<td>
Open multiple tabs becomes slow
</td>
</tr>
</tbody>
</table>
</p>






## Comment example - [bug 1795009](https://bugzilla.mozilla.org/show_bug.cgi?id=1795009)


This bug was moved into the Performance component.

:zero.arst, could you make sure the following information is on this bug?

- ~~For slowness or high CPU usage, capture a profile with http://profiler.firefox.com/, upload it and share the link here.~~
- For memory usage issues, capture a memory dump from `about:memory` and attach it to this bug.
- Troubleshooting information: Go to `about:support`, click "Copy raw data to clipboard", paste it into a file, save it, and attach the file here.


If the requested information is already in the bug, please confirm that.

Thank you.

**Comment by suhaibmujahid** [comment]
@julienw in the context of doing some checks to avoid asking for information that is already present in the bug, do you think we should consider only content that was submitted be the reporter?

File: auto_nag/scripts/moved_to_performance.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                "application/json",
+                "text/plain",
+            )
+            or "memory-report" in attachment["file_name"]
+        ):
+            return False
+
+        content_bytes = base64.b64decode(attachment["data"])
+        try:
+            content = json.loads(content_bytes)
```
Comment (marco-c) [inline auto_nag/scripts/moved_to_performance.py:85]: Sometimes people attach textual info, which looks like:
```
Informazioni di base
--------------------

Nome: Firefox
Versione: 112.0a1
ID build: 20230312211644
...

Segnalazioni di arresto anomalo degli ultimi 3 giorni
-----------------------------------------------------
...
```

Not sure if we can easily detect it, as it's translated.
Maybe we could just check if there are multiple lines with multiple `-` characters.

**Comment by marco-c** [comment]
I never used a checklist on Bugzilla, does it actually work? Can a user mark something in a checklist as done? Do they need specific permissions to do that if the comment is from somebody else (the bot in this case)?

**Comment by julienw** [comment]
> @julienw in the context of doing some checks to avoid asking for information that is already present in the bug, do you think we should consider only content that was submitted be the reporter?

Yes, this sounds reasonable.
I wonder if we should also have a time limit. For example if the information is older than say 6 months, we could assume this is outdated and still ask about it.

**Comment by suhaibmujahid** [comment]
>  I wonder if we should also have a time limit. For example if the information is older than say 6 months, we could assume this is outdated and still ask about it.

Done in 2a7cae1fedb6b20c046711f564b5df0a82f47bd6

**Comment by suhaibmujahid** [comment]
> I never used a checklist on Bugzilla, does it actually work? Can a user mark something in a checklist as done? Do they need specific permissions to do that if the comment is from somebody else (the bot in this case)?

Unfonratintly, Bugzilla does not support markdown checklists.

File: auto_nag/scripts/moved_to_performance.py (hunk 1 above)
Comment (suhaibmujahid) [reply auto_nag/scripts/moved_to_performance.py:96]: The comment will require clinking on "Copy **raw** data to clipboard", not "Copy **text** to clipboard". So, it should be fine if we ask for the data if it is not matching the expected format.

WDYT?

Comment (julienw) [reply auto_nag/scripts/moved_to_performance.py:103]: Yes this works for me. If we see this brings problems we can adjust later.

**Comment by marco-c** [comment]
Maybe when the data is present but not recent enough, we should adjust the comment to ask to refresh the info (otherwise people will ignore it and say that it is already present)

**Comment by suhaibmujahid** [comment]
> Maybe when the data is present but not recent enough, we should adjust the comment to ask to refresh the info (otherwise people will ignore it and say that it is already present)

Do you think 13820a756713e2715b82a0e5ecac7f4fb7a80d81 would be enough? If not, I can implement what you suggested.

**Pull request merged**